File: Intermediate/11.Convert a decimal number to binary/Solution.java
import java.util.Scanner;

public class DecimalToBinary {
    public static void decimalToBinary(int n) {
        if (n == 0) {
            System.out.print("0");
            return;
        }

        String binary = "";
        while (n > 0) {
            binary = (n % 2) + binary;
            n /= 2;
        }

        System.out.print(binary);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter a decimal number: ");
        int num = scanner.nextInt();

        System.out.print("Binary representation: ");
        decimalToBinary(num);
        System.out.println();
    }
}

File: Intermediate/11.Convert a decimal number to binary/Solution.py
def decimal_to_binary(n):
    if n == 0:
        return "0"

    binary = ""
    while n > 0:
        binary = str(n % 2) + binary
        n //= 2

    return binary

if __name__ == "__main__":
    num = int(input("Enter a decimal number: "))
    print("Binary representation:", decimal_to_binary(num))
